_posts/2023/2023-02-13-try-not-to-loop-through-associations-inside-jobs.md

```diff
@@ -0,0 +1,74 @@
+---
+title: "Try not to loop through associations inside jobs"
+description: "Your local data isn’t the same as in production"
+layout: article
+category: ruby
+image:
+  base: '2022/try-not-to-loop-through-associations-inside-jobs'
+  alt: "Roller coaster loops"
+  credit: "Priscilla Du Preez"
+  source: "https://unsplash.com/photos/FOsina4f7qM"
+
+---
+
+One of the downside of using an level of abstraction above your database is that it’s easy to accidentally give yourself performance headaches via running multiple very similar queries.
+
+This is typically referred to as an n+1 problem, which is most often seen when you display an Active Record object in a view and show all of the associated records that `belong_to` to it. We often solve this by putting heavier database work in asynchronous jobs.
```

Comment:
*N+1

*problem. They occur most often when you...

*...They tend to be solved by

Reply:
```suggestion
This is typically referred to as an N+1 problem. they occur most often seen when you display an Active Record object in a view and show all of the associated records that `belong_to` to it. They tend to be solved by putting heavier database work in asynchronous jobs.
```